rust/extns/numaflow-sqs/src/source.rs

```diff
@@ -26,12 +25,104 @@ pub const SQS_DEFAULT_REGION: &str = "us-west-2";
 /// Configuration for an SQS message source.
 ///
 /// Used to initialize the SQS client with region and queue settings.
-/// Implements serde::Deserialize to support loading from configuration files.
-/// TODO: add support for all sqs configs and different ways to authenticate
-#[derive(serde::Deserialize, Clone, PartialEq)]
+/// TODO: use config in sqs actor methods
+#[derive(Debug, Clone, PartialEq)]
 pub struct SQSSourceConfig {
+    // Required fields
     pub region: String,
     pub queue_name: String,
+    pub auth: SQSAuth,
+
+    // Optional fields
+    pub visibility_timeout: Option<i32>,
+    pub max_number_of_messages: Option<i32>,
+    pub wait_time_seconds: Option<i32>,
+    pub endpoint_url: Option<String>,
+    pub attribute_names: Vec<String>,
+    pub message_attribute_names: Vec<String>,
+}
+
+impl SQSSourceConfig {
+    /// Validates the SQS source configuration
+    pub fn validate(&self) -> Result<()> {
+        // Validate required fields
+        if self.region.is_empty() {
+            return Err(Error::InvalidConfig("region is required".to_string()));
```

Comment:
Some validations can be moved to controller side, other validations can be done before creating the config.